ext/rapidjson/parser.hh

```diff
@@ -86,9 +95,30 @@ struct RubyObjectHandler : public BaseReaderHandler<UTF8<>, RubyObjectHandler> {
         return true;
     }
 
-    bool push(VALUE val) {
+    void materialize_hash() {
+        auto top_type = stack_type[depth - 1];
+
+        if (top_type == ObjectType::BufferedHash) {
+            if (hash_buffer_idx & 1) {
+                // drop last key
+                hash_buffer_idx--;
+            }
+
+            VALUE hash = rb_hash_new_capa(hash_buffer_idx / 2);
```

Comment:
I added `rb_hash_new_capa` in 3.2, we'd likely need a fallback for older rubies.